crates/plugins/src/manifest.rs

```diff
@@ -137,7 +137,9 @@ fn inner_check_supported_version(
     })?;
     let version = Version::parse(spin_version)?;
     if !comparator.matches(&version) {
-        if override_compatibility_check {
+        if !version.pre.is_empty() {
```

Comment:
@kate-goldenring @karthik2804 The `semver` crate is very picky about pre-releases, which probably makes sense given how they are treated in the Semantic Versioning spec.  I'm working around it based on the idea that a pre-release has unknown compatibility constraints in general, so anything goes.  Let me know if you think there's a better approach.